www/include/configuration/configObject/traps/listTraps.php

```diff
@@ -217,6 +243,8 @@ function setO(_i) {
 
 $tpl->assign('limit', $limit);
 $tpl->assign('searchT', $search);
+$tpl->assign('searchVendor', $searchVendor);
+$tpl->assign('searchStatus', (string)$searchStatus);
```

Comment:
line 58, you're sanitizing the variable and searching for an integer. Why casting a string here ?

Reply:
it needs to be string in smarty, because there is status with value 0 and if the var is not string - it is always selected.